Tribler/Core/Session.py

```diff
@@ -648,6 +651,41 @@ def create_channel(self, name, description, mode=u'closed'):
         """
         return self.lm.channel_manager.create_channel(name, description, mode)
 
+    @blocking_call_on_reactor_thread
```

Comment:
Ok, I had a look and the reason it's failing is because you are calling twisted stuff from the nose thread on the unit tests. You should either decorate the tests or make a decorated inner method.